resource_leak.py

```diff
@@ -0,0 +1,44 @@
+# DB resource leak demo
+import psycopg2
+
+def test_connection():
+    connection = psycopg2.connect(user="postgres",
+                                  password="postgres123",
+                                  host="127.0.0.1",
+                                  port="5432",
+                                  database="postgres_db")
+    cursor = connection.cursor()
+    cursor.execute("SELECT version();")
+    record = cursor.fetchone()
+
+# File resource leak demo
+def read_write_data(data):
+    file = open("test-file.txt", "r")
+    out = file.read()
+    file.write(out)
+
+
+def read_lines(file):
+    lines = []
+    f = open(file, "r")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *f*. The resource is allocated by call *builtins.open*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close *f* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).